pallets/staking/src/lib.rs

```diff
@@ -1091,12 +1097,14 @@ decl_module! {
             let stash_balance = <T as Trait>::Currency::free_balance(&stash);
             let value = value.min(stash_balance);
             let item = StakingLedger {
-                stash,
+                stash: stash.clone(),
                 total: value,
                 active: value,
                 unlocking: vec![],
                 last_reward: Self::current_era(),
             };
+            let did = Context::current_identity::<T::Identity>().unwrap_or_default();
+            Self::deposit_event(RawEvent::Bonded(did, stash, value));
```

Comment:
Emitting the controller address will also be good?

Reply:
Events are mainly used to maintain historic views. I don't think having the controller address is much useful in historic graphs and such. FWIW all of this info and the controller can be directly queried from the chain anyway.